driver/docker/docker.go

```diff
@@ -414,29 +413,31 @@ func (d *Driver) inspectImage(ctx context.Context, image bundle.InvocationImage)
 // validateImageDigest validates the operation image digest, if exists, against
 // the supplied repoDigests
 func (d *Driver) validateImageDigest(image bundle.InvocationImage, repoDigests []string) error {
-	if image.Digest != "" {
-		switch count := len(repoDigests); {
-		case count == 0:
-			return fmt.Errorf("image %s has no repo digests", image.Image)
-		case count > 1:
-			return fmt.Errorf("image %s has more than one repo digest", image.Image)
-		}
+	if image.Digest == "" {
+		return nil
+	}
 
-		repoDigest := repoDigests[0]
-		// RepoDigests are of the form 'imageName@sha256:<sha256>'; we parse out the digest itself for comparison
-		// TODO: is there a better (i.e. via docker api) way to extract the digest?
-		parts := strings.Split(repoDigest, "@")
-		if len(parts) != 2 {
-			return fmt.Errorf("unable to parse repo digest %s", repoDigest)
-		}
-		digest := parts[1]
+	switch count := len(repoDigests); {
+	case count == 0:
+		return fmt.Errorf("image %s has no repo digests", image.Image)
+	case count > 1:
+		return fmt.Errorf("image %s has more than one repo digest", image.Image)
+	}
 
-		if digest == image.Digest {
-			return nil
-		}
-		return fmt.Errorf("content digest mismatch: image %s has digest %s but the value should be %s according to the bundle file", image.Image, digest, image.Digest)
+	// RepoDigests are of the form 'imageName@sha256:<sha256>'; we parse out the digest itself for comparison
+	repoDigest := repoDigests[0]
+	ref, err := reference.ParseNormalizedNamed(repoDigest)
+	if err != nil {
+		return err
```

Comment:
Sorry I gave you informal code. We should probably wrap this error and provide a better error message with context. Otherwise it won't be clear what reference is invalid.

Reply:
Updated!  And deadbeef returns in a test for this scenario! 🎉